File: main/src/main/java/net/fwitz/math/main/binary/complex/functions/examples/simple/JedSoftExample1.java
package net.fwitz.math.main.binary.complex.functions.examples.simple;

import net.fwitz.math.binary.complex.Complex;
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot;

import static java.lang.Math.PI;

// http://www.jedsoft.org/fun/complex/
public class JedSoftExample1 {
    public static final String FN_NAME = "f(z) = [z + z^2 / sin(z^4 - 1)]^2";

    public static void main(String[] args) {
        new ComplexFunctionPlot(FN_NAME)
                .domainX(-PI, PI)
                .domainY(-PI, PI)
                .computeFn(JedSoftExample1::fn)
                .render();
    }

    public static Complex fn(Complex z) {
        Complex z2 = z.times(z);
        Complex z4 = z2.times(z2);
        Complex frac = z2.div(z4.minus(1).sin());
        Complex brackets = z.plus(frac);
        return brackets.times(brackets);
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/complex/functions/examples/simple/JedSoftExample1.kt
package net.fwitz.math.main.binary.complex.functions.examples.simple

import net.fwitz.math.binary.complex.Complex
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot

// http://www.jedsoft.org/fun/complex/
object JedSoftExample1 {
    const val FN_NAME = "f(z) = [z + z^2 / sin(z^4 - 1)]^2"

    @JvmStatic
    fun main(args: Array<String>) = ComplexFunctionPlot(FN_NAME)
        .domainX(-Math.PI, Math.PI)
        .domainY(-Math.PI, Math.PI)
        .computeFn(::fn)
        .render()

    fun fn(z: Complex): Complex {
        val z2 = z.pow2
        val z4 = z2.pow2
        val brackets = z + z2 / (z4 - 1).sin
        return brackets.pow2
    }
}